src/renderer/components/ReactFlowBox.tsx

```diff
@@ -297,25 +326,35 @@ export const ReactFlowBox = memo(({ wrapperRef, nodeTypes, edgeTypes }: ReactFlo
                         targetControlY,
                         targetX,
                         targetY,
-                    });
-                }
-
-                // If not is not in the axis-aligned bounding box of the edge, we can skip it
-                const aabb = {
-                    minX: Math.min(edgeLine.sourceX, edgeLine.targetX),
-                    maxX: Math.max(edgeLine.sourceX, edgeLine.targetX),
-                    minY: Math.min(edgeLine.sourceY, edgeLine.targetY),
-                    maxY: Math.max(edgeLine.sourceY, edgeLine.targetY),
-                };
-                if (
-                    node.position.x > aabb.maxX || // node is to the right of the edge
-                    node.position.x + (node.width || 0) < aabb.minX || // node is to the left of the edge
-                    node.position.y > aabb.maxY || // node is below the edge
-                    node.position.y + (node.height || 0) < aabb.minY // node is above the edge
-                ) {
-                    return false;
+                    ];
+                    const curve = new Bezier(coords);
+                    const curveIntersectsLeft =
+                        curve.lineIntersects({
+                            p1: nodeBounds.TL,
+                            p2: nodeBounds.BL,
+                        }).length > 0;
+                    const curveIntersectsRight =
```

Comment:
Another note to myself: i can just do early returns here to avoid doing unnecessary calculations. There's probably also a way i can check which side is closest to the curve and check that first.

Reply:
Another idea: we can store the center point of the curve (from the edge data) and check if the center is within the node bounds. It would definitely mean collision and would be much faster to check.